mail_tracking/models/mail_message.py

```diff
@@ -95,6 +105,18 @@ def _get_email_cc(self):
             })
         return res
 
+    @api.multi
+    def _get_failed_message(self):
+        res = {}
+        for message in self:
+            res.update({
+                message.id: message.track_needs_action
+                and any(message.mail_tracking_ids.filtered(
```

Comment:
I think this is not properly done.

Correct me if I'm wrong, but the `track_needs_action` by itself already indicates that the state is not good, right? In such case, the next check wouldn't be needed.

Reply:
Perhaps can need an action but not related with 'failed message'

Reply:
OK, then just a little change:

```suggestion
                and bool(message.mail_tracking_ids.filtered(
```